program/rust/src/processor/upd_price.rs

```diff
@@ -175,21 +176,82 @@ pub fn upd_price(
         )?;
     }
 
-    // Try to update the aggregate
-    #[allow(unused_variables)]
-    let mut aggregate_updated = false;
+    {
+        // Reload price data as a struct after c_upd_aggregate() borrow is dropped
+        let mut price_data = load_checked::<PriceAccount>(price_account, cmd_args.header.version)?;
+
+        // Try to update the publisher's price
+        if is_component_update(cmd_args)? {
```

Comment:
mmmmmm moving this up here seems complicated, maybe this should be a different PR?

Reply:
I'm worried about the side effects of this